Readbytesb -- return bytes object for efficiency

aardvarkkrill opened the pull request:
wow ... my first ever pull request!  Also my first ever python C-api foray.
So apologies if I've done anything stupid...

added readbytesb which returns a bytes object rather than a list.  This can, for example be fed directly into numpy.frombuffer giving more efficient transfer as no list needs to be built.

doceme commented on spidev_module.c:
```diff
@@ -223,27 +223,41 @@ SpiDev_writebytes(SpiDevObject *self, PyObject *args)
+	Py_buffer pybuff;
+	uint8_t* rxbuf = stackbuf;
+
+    if (resultType == 2) { //existing bytearray
```
Please remove magic numbers (Add #define for `resultType`)

aardvarkkrill commented:
Good call on the magic numbers.  I think perhaps my refactoring is a bit heavy handed anyway, and I've changed it a bit.
I added another function (which is probably the one I should have written first) and I'll make a separate pull request for that... again I'm new here, sorry if this is wrong.

fschrempf commented:
Superseded by https://github.com/doceme/py-spidev/pull/83
